Bottom: bdd5d6076351dd75f4bf03fd7941bd0d77b21941
Top:    f09c2ad6f5b5eb566be63f53c50d1b6f156da629
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-04-25 15:24:09 -0500

Wait on openldap container startup

Occasionally the openldap container would not be ready when attempting
to add the admin and proxyagent accounts. A wait is required to
ensure the service is running properly.


---
diff --git a/roles/docker-openldap/tasks/main.yml b/roles/docker-openldap/tasks/main.yml
index e771b77..35c5ba7 100644
--- a/roles/docker-openldap/tasks/main.yml
+++ b/roles/docker-openldap/tasks/main.yml
@@ -149,6 +149,13 @@
 # *****************************************************************************
 # Set the LDAP passwords
 
+- name: wait for LDAP to start
+  wait_for:
+    delay: 5
+    host: '{{ container_addr_map.ldap.ip_addr }}'
+    port: 389
+    state: started
+
 - name: set LDAP admin password
   command: 'docker exec -i {{ openldap_container_name }} /docker-entrypoint.sh set_admin_pass {{ ldap_admin_password | quote }}'
